### PR TITLE
SCRUM-2338 Load and display allele_database_status attribute

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

**Review by adamgibs — APPROVED** [review]
UI stuff looks good

File: src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/alleleSlotAnnotations/AlleleDatabaseStatusSlotAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@Audited
+@Entity
+@Data
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
+@Inheritance(strategy = InheritanceType.JOINED)
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/alleleSlotAnnotations/AlleleDatabaseStatusSlotAnnotation.java]: This (`@Inheritance(strategy = InheritanceType.JOINED)`) only needs to be on here if there is a inheriting subclass.

**Review by oblodgett — CHANGES_REQUESTED** [review]
Still have some other checks to do.

File: src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+ALTER TABLE alleledatabasestatusslotannotation_aud ADD CONSTRAINT alleledatabasestatus_aud_id_rev_fk FOREIGN KEY (id, rev) REFERENCES slotannotation_aud (id, rev);
+
+INSERT INTO vocabulary (id, name) VALUES (nextval('hibernate_sequence'), 'Allele database status vocabulary');
+	
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'approved', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'deleted', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'reserved', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'autoloaded', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
+INSERT INTO vocabularyterm (id, name, vocabulary_id) SELECT nextval('hibernate_sequence'), 'in_progress', id FROM vocabulary WHERE name = 'Allele database status vocabulary';
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql:31]: Also looks like there are two missing FK's:

```
ALTER TABLE public.alleledatabasestatusslotannotation_aud
    ADD CONSTRAINT fko47ewqwil249eqgi82tcc3wxn FOREIGN KEY (id, rev) REFERENCES public.nameslotannotation_aud(id, rev);
```

and

```
ALTER TABLE public.alleledatabasestatusslotannotation
    ADD CONSTRAINT fkbjl9t9iwmxl7jgxuwpi8r3atc FOREIGN KEY (id) REFERENCES public.nameslotannotation(id);
```
   
Always need to test against what hibernate is going to do.

**Review by oblodgett — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql (hunk 2 above)
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql:31]: Actually, it was the class definition that was wrong - it shouldn't be extending NameSlotAnnotation, just SlotAnnotation

File: src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/alleleSlotAnnotations/AlleleDatabaseStatusSlotAnnotation.java (hunk 1 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/alleleSlotAnnotations/AlleleDatabaseStatusSlotAnnotation.java]: Will remove from this and other allele/gene slot annotation classes

File: src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql (hunk 2 above)
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql:31]: Was missing an FK constraint on the id field to the slotAnnotation table though - fixed now

File: src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/alleleSlotAnnotations/AlleleDatabaseStatusSlotAnnotation.java (hunk 1 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/alleleSlotAnnotations/AlleleDatabaseStatusSlotAnnotation.java]: Removed

File: src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql (hunk 2 above)
Comment (oblodgett) [reply src/main/resources/db/migration/v0.21.0.1__add_allele_database_status.sql:31]: Ahh ok.

**Review by oblodgett — APPROVED** [review]

**Comment by oblodgett** [comment]
PS... I have found some duplicate indexing on some of the tables on alpha. This would cause slower insert times, if you have a few maybe you could go through some of the slot annotation tables and create a migration to remove some of the extra (duplicate) indexes. But I would suggest that in a separate PR.

**Comment by markquintontulloch** [comment]
Sure, will take a look at them in another PR

**Pull request merged**